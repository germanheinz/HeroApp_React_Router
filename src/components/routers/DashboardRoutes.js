import React from 'react'
import { Route, Switch } from 'react-router'
import { DcScreen } from '../dc/DcScreen'
import { LoginScreen } from '../login/LoginScreen'
import { MarvelScreen } from '../marvel/MarvelScreen'
import { Navbar } from '../ui/Navbar'

export const DashboardRoutes = () => {
    return (
        <>

        <Navbar />
        <div>
            <Switch>
                {/* <Route exact path="/login" component={ LoginScreen } /> */}
                <Route exact path="/marvel" component={ MarvelScreen } />
                <Route exact path="/dc" component={ DcScreen } />
            </Switch>
        </div>
            
        </>
    )
}
