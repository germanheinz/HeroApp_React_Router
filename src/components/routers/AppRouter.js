import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from "react-router-dom";
import { LoginScreen } from '../login/LoginScreen';
import { Navbar } from '../ui/Navbar';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {
    return (
     <Router>
      <div>
        <Switch>
           <Route exact path="/login" component={ LoginScreen } />
           <Route path="/" component={ DashboardRoutes } />
        </Switch>

      </div>
    </Router>
  );
}
