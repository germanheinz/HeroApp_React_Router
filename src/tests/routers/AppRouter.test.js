
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { AppRouter } from '../../components/routers/AppRouter';
import { AuthContext } from '../../components/auth/AuthContext';

Enzyme.configure({ adapter: new Adapter() });

describe('Pruebas en <AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }
    
    test('debe mostrar login si no esta autenticado', () => {
        
        

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )


        // console.log(wrapper.html() );

        expect( wrapper ).toMatchSnapshot();


    })

    test('debe mostrar componente marvel si estÃ¡ autenticado', () => {
        
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'german'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper.find('.navbar').exists()).toBe(true);

    })
    
    

})