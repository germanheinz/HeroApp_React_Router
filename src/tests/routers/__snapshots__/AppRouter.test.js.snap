// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe mostrar login si no esta autenticado 1`] = `ReactWrapper {}`;
